doc/reference/cardano-governance.md

```diff
@@ -0,0 +1,235 @@
+
+  * If it becomes a _candidate_, **later** than 4k slots before the end of the epoch, the adoption is delayed 1 epoch until the transition to _Epoch N+2_, provided it becomes an _stable candidate_.
+
+6. **Stable candidate**: The last required endorsement is 2k slots deep.
+
+  If there was no _stable candidate_ proposal by the end of the epoch where the proposal was submitted, it is carried over to the next epoch; in particular, a candidate proposal whose threshold-satisfying endorsement was not yet stable will be adopted at the subsequent epoch unless it is surpassed (by a new proposal) in the meantime.
+
+7. **Update**: Update proposals are adopted at epoch transitions.
+
+### How to create a Byron era update proposal
+
+Cardano-cli provides the necessary tools to create and submit update proposals and votes:
+
+```bash
+cardano-cli byron governance
+Usage: cardano-cli byron governance COMMAND
+
+  Byron governance commands
+
+Available commands:
+  create-update-proposal   Create an update proposal.
+  create-proposal-vote     Create an update proposal vote.
+  submit-update-proposal   Submit an update proposal.
+  submit-proposal-vote     Submit a proposal vote.
+
+```
+Here is an example of an update proposal to go from Byron (Ouroboros BFT) to Shelley (TPraos). We increase the protocol version from 1.0.0 to 2.0.0:
+
+```bash
+cardano-cli byron governance create-update-proposal \
+--filepath update.proposal \
+--testnet-magic 42 \
+--signing-key byron.delegate.key \
+--protocol-version-major "2" \
+--protocol-version-minor "0" \
+--protocol-version-alt "0" \
+--application-name "cardano-sl" \
+--software-version-num "1" \
+--system-tag "linux" \
+--installer-hash 0
+```
+Then, to submit the update proposal:
+
+```bash
+cardano-cli byron governance submit-update-proposal \
+--testnet-magic 42 \
+--filepath update.proposal
+```
+ Voting works in a similar way by creating the vote:
+
+```bash
+cardano-cli byron governance create-proposal-vote \
+--proposal-filepath update.proposal \
+--testnet-magic 42 \
+--signing-key byron.delegate.key \
+--vote-yes \
+--output-filepath update.proposal.vote
+```
+and submitting it:
+
+```bash
+cardano-cli byron governance submit-proposal-vote  \
+--testnet-magic 42 \
+--filepath update.proposal.vote
+```
+In this case, delegate nodes must be running a node version that is compatible with the protocol version 2.0 to endorse the update.
+
+## Update proposals in Shelley+ eras
+
+There are some changes in the update proposal mechanism in Shelley and later eras: there is no distinction between votes and proposals. To 'vote' for a proposal, a _delegate_ submits the exact same update proposal. Update proposals and votes are submitted in a regular transaction. Also, there is no separate endorsement step.
+
+The procedure is as follows:  
+
+![](../reference/images/shelleyup.png)
```

Comment:
this picture is fantastic!  would it help to mention that the entire epoch has 10k/f many slots?